### DISCUSSION TITLE
weight_name = 'distance'

### DISCUSSION THREAD

**Discussion by Adriano-Silva-DEV** [Project-OSRM/osrm-backend#6801] [Q&A]
Hello, has anyone had problems selecting weight_name = 'distance'?
I have a problem with a route, 2 points close to each other, but the route always has another city to return to later.

**Comment by frodrigo** [comment]
If you still have the issue. Please, reformulate it to help understand what do you mean.

**Reply by Adriano-Silva-DEV** [reply]
I have the following issue: when using the Weight_name = 'distance' parameter, in some routes where there are very close points in a city, for example, two points with a distance of 50 meters between them, it should choose the closest point to follow the route. However, it ends up going to another city and then returning to the nearby point.

**Reply by Adriano-Silva-DEV** [reply]
![Captura de tela 2024-03-05 092958](https://github.com/Project-OSRM/osrm-backend/assets/42922775/859e1e6f-093a-4fcb-8c07-c0d140985f07)

**Comment by frodrigo** [comment]
Do you have the coordinates of the points ? Do you check the topology of
OSM data ?

**Reply by Adriano-Silva-DEV** [reply]
lat: -9.74946687209306, lng: -37.43329165979751 , lat: -9.74946, lng: -37.43382

**Reply by frodrigo** [reply]
https://www.openstreetmap.org/directions?engine=fossgis_osrm_car&route=-9.74948%2C-37.43331%3B-9.74944%2C-37.43389

Topology look good on demo server.

The issue is only with distance ?

**Reply by Adriano-Silva-DEV** [reply]
I think I didn't explain it correctly. These two points are part of a route. I am using the OSRM service to generate the route, and OSRM itself to determine the best path and order of the points. Therefore, these two points in the list should follow in order, as they are very close, and there is no reason for the route to go to another city and then come back.

**Reply by Adriano-Silva-DEV** [reply]
This is my list of coordinates (points).
 [
				{
					"id": "1125",
					"latitude": -9.682895,
					"longitude": -36.303805	
				},
				{
					"id": "1651",
					"latitude": -9.645186,
					"longitude": -36.49655				
				},
				{
					"id": "157",
					"latitude": -9.647881020089354,
					"longitude": -36.49263267329154			
				},
				{
					"id": "1097",
					"latitude": -9.674099143459005,
					"longitude": -37.12425572715474				
				},
				{
					"id": "557",
					"latitude": -9.674392834236436,
					"longitude": -37.124146230962765		
				},
				{
					"id": "1347",
					"latitude": -9.681359564949078,
					"longitude": -37.12594540323796				
				},
				{
					"id": "1142",
					"latitude": -9.673167955492735,
					"longitude": -37.12938690204803			
				},
				{
					"id": "1460",
					"latitude": -9.526999,
					"longitude": -37.295795	
				},
				{
					"id": "761",
					"latitude": -9.536489206953618,
					"longitude": -37.29398148863526					
				},
				{
					"id": "1105",
					"latitude": -9.5355625,
					"longitude": -37.2944969					
				},
				{
					"id": "1104",
					"latitude": -9.553387641863313,
					"longitude": -37.37793523096391			
				},
				{
					"id": "1618",
					"latitude": -9.554085,
					"longitude": -37.380203
				},
				{
					"id": "1102",
					"latitude": -9.559447705265766,
					"longitude": -37.37943679312136	
				},
				{
					"id": "2202",
					"latitude": -9.74946,
					"longitude": -37.43382					
				},
				{
					"id": "711",
					"latitude": -9.74946687209306,
					"longitude": -37.43329165979751
				}]
